Hashtable_.c

```diff
@@ -0,0 +1,197 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#define CAPACITY 50000 // Size of the HashTable.
+
+unsigned long hash_function(char *str)
+{
+    unsigned long i = 0;
+
+    for (int j = 0; str[j]; j++)
+        i += str[j];
+
+    return i % CAPACITY;
+}
+
+// Defines the HashTable item.
+typedef struct Ht_item
+{
+    char* key;
+    int value;
+    struct Ht_item* next;
+} Ht_item;
+
+
+
+// Defines the HashTable
+typedef struct HashTable
+{
+    // Contains an array of pointers to items.
+    Ht_item **items;
+    int size;
+    int count;
+} HashTable;
+
+
+Ht_item *create_item(char* key,int value)
+{
+    // Creates a pointer to a new HashTable item.
+    Ht_item *item = (Ht_item *)malloc(sizeof(Ht_item));
+    item->key = malloc(strlen(key)+1);
+    strcpy(item->key, key); 
+    item->value = value;
+    item->next = NULL;
+    return item;
+}
+
+HashTable *create_table(int size)
+{
+    // Creates a new HashTable.
+    HashTable *table = (HashTable *)malloc(sizeof(HashTable));
+    table->size = size;
+    table->count = 0;
+    table->items = (Ht_item **)malloc(table->size * sizeof(Ht_item *));
+
+    for (int i = 0; i < table->size; i++)
+        table->items[i] = NULL; 
+    return table;
+}
+
+
+void ht_insert(HashTable *table, char *key, int value)
```

Comment:
maybe change that the function will return bool that you will know if you succeeded.